omnichain/swap/contracts/SwapToZeta.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.7;
+
+import "@zetachain/protocol-contracts/contracts/zevm/SystemContract.sol";
+import "@zetachain/protocol-contracts/contracts/zevm/interfaces/zContract.sol";
+import "@zetachain/toolkit/contracts/SwapHelperLib.sol";
+import "@zetachain/toolkit/contracts/BytesHelperLib.sol";
+import "@zetachain/protocol-contracts/contracts/zevm/interfaces/IWZETA.sol";
+
+contract SwapToZeta is zContract {
+    SystemContract public systemContract;
+
+    uint256 constant BITCOIN = 18332;
+
+    constructor(address systemContractAddress) {
+        systemContract = SystemContract(systemContractAddress);
+    }
+
+    modifier onlySystem() {
+        require(
```

Comment:
in this case I understand it's consistent to use require, but in general we try to use custom errors. Can us use this pr to change both? this new one and the same modifier in the other file?

Reply:
Something like this?

```solidity
    error OnlySystemContract(string);

    modifier onlySystem() {
        if (msg.sender != address(systemContract)) {
            revert OnlySystemContract(
                "Only system contract can call this function"
            );
        }
        _;
    }
```

Instead of having this in the template, how about we make the modifier a part of another contract and import it?

Reply:
If we don't want to include it in the `zContract`, maybe like this?

```solidity
contract Swap is zContract, OnlySystem {
  ...
}
```

@andresaiello

Reply:
even better

Reply:
@andresaiello do we want to keep the `OnlySystem` modified in protocol contracts or toolkit? Probably toolkit, right?

Reply:
https://github.com/zeta-chain/toolkit/pull/131

Reply:
Let's merge and ship this in v8 and then I'll remove the modifier from example contracts.